sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala

```diff
@@ -97,7 +100,12 @@ object HiveResult {
     case (null, _) => "null"
     case (s: String, StringType) => "\"" + s + "\""
     case (decimal, DecimalType()) => decimal.toString
-    case (interval, CalendarIntervalType) => interval.toString
+    case (interval: CalendarInterval, CalendarIntervalType) =>
+      SQLConf.get.intervalOutputStyle match {
+        case SQL_STANDARD => toSqlStandardString(interval)
+        case ISO_8601 => toIso8601String(interval)
+        case _ => toMultiUnitsString(interval)
```

Comment:
shall we be Hive compatible and always return `toMultiUnitsString` here?

Reply:
1. we may need to escape some tests for `ThriftServerQueryTestSuite`, does it worth to do it separately?
2.  IMO, add a new config here is enough to be Hive compatible